diff --git a/node_modules/react-joyride/es/components/Portal.js b/node_modules/react-joyride/es/components/Portal.js
index 7a57a07..8e49d60 100644
--- a/node_modules/react-joyride/es/components/Portal.js
+++ b/node_modules/react-joyride/es/components/Portal.js
@@ -1,5 +1,5 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
+import { createPortal } from 'react-dom';
 import PropTypes from 'prop-types';
 import { isReact16 } from '../modules/helpers';
 
@@ -10,31 +10,13 @@ function Portal(props) {
     return null;
   }
 
-  if (isReact16()) {
-    return ReactDOM.createPortal(children, portal);
-  }
-
-  return /*#__PURE__*/React.createElement(React.Fragment, null, children);
-}
-
-Portal.prototype.componentDidMount = function componentDidMount() {
-  if (!isReact16()) {
-    this.renderPortal();
-  }
-};
-
-Portal.prototype.componentDidUpdate = function componentDidUpdate() {
-  if (!isReact16()) {
-    this.renderPortal();
-  }
-};
-
-Portal.prototype.componentWillUnmount = function componentWillUnmount() {
-  if (!isReact16()) {
-    ReactDOM.unmountComponentAtNode(this.node);
-  }
-};
-
-Portal.prototype.renderPortal = function renderPortal() {
-  ReactDOM.unstable_renderSubtreeIntoContainer(this, this.props.children, this.node);
-};
+  return createPortal(children, portal);
+}
 
 Portal.propTypes = {
   children: PropTypes.element.isRequired,
   portal: PropTypes.oneOfType([PropTypes.object, PropTypes.string])
 };
 export default Portal;
+
diff --git a/node_modules/react-joyride/lib/components/Portal.js b/node_modules/react-joyride/lib/components/Portal.js
index b8ac4ce..10c2662 100644
--- a/node_modules/react-joyride/lib/components/Portal.js
+++ b/node_modules/react-joyride/lib/components/Portal.js
@@ -10,7 +10,7 @@
 
 var _react = _interopRequireDefault(require("react"));
 
-var _reactDom = _interopRequireDefault(require("react-dom"));
+var _reactDom = require("react-dom");
 
 var _propTypes = _interopRequireDefault(require("prop-types"));
 
@@ -23,37 +23,13 @@
     return null;
   }
 
-  if ((0, _helpers.isReact16)()) {
-    return _reactDom.default.createPortal(children, portal);
-  }
-
-  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, children);
-}
-
-Portal.prototype.componentDidMount = function componentDidMount() {
-  if (!(0, _helpers.isReact16)()) {
-    this.renderPortal();
-  }
-};
-
-Portal.prototype.componentDidUpdate = function componentDidUpdate() {
-  if (!(0, _helpers.isReact16)()) {
-    this.renderPortal();
-  }
-};
-
-Portal.prototype.componentWillUnmount = function componentWillUnmount() {
-  if (!(0, _helpers.isReact16)()) {
-    _reactDom.default.unmountComponentAtNode(this.node);
-  }
-};
-
-Portal.prototype.renderPortal = function renderPortal() {
-  _reactDom.default.unstable_renderSubtreeIntoContainer(this, this.props.children, this.node);
-};
+  return (0, _reactDom.createPortal)(children, portal);
+}
 
 Portal.propTypes = {
   children: _propTypes.default.element.isRequired,
   portal: _propTypes.default.oneOfType([_propTypes.default.object, _propTypes.default.string])
 };
 var _default = Portal;
 exports.default = _default;
+
